src/Compilers/CSharp/Test/Emit2/Emit/RuntimeProbing/ModuleCancellationTests.cs

```diff
@@ -2783,4 +2783,28 @@ void G<S>(T a, S b) {}
         var verifier = CompileAndVerify(source);
         AssertNotInstrumentedWithTokenLoad(verifier, "C<T>.G<S>(T, S, System.Threading.CancellationToken)");
     }
+
+    [Fact]
+    public void FlowPass()
+    {
+        var source = """
+            using System.Threading;
+            using System.Collections.Generic;
+
+            class C
+            {
+                IEnumerable<int> F()
+                {
+                    var x = G() as string;
+                    yield return (x != null) ? 1 : 0;
+                }
+
+                object G(CancellationToken token = default)
+                    => "";
+            }
+            """;
+
+        // definite assignment flow pass doesn't fail
+        _ = CompileAndVerify(source);
```

Comment:
We prefer to ensure that warnings are also asserted, which they won't be by just `CompileAndVerify`.

```suggestion
        CompileAndVerify(source).VerifyDiagnostics();
```